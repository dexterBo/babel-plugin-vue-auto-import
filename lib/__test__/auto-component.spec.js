const babel = require('@babel/core');
const _ = require('lodash');
const plugin = require('../index');
const makeStore = require('./__mocks__/store.mock');

jest.mock('vue-template-label-loader/lib/store', () => require('./__mocks__/store.mock'));

const example = `
import { Button } from 'element-ui';

export default {
  name: 'About',
  components: {
    [Button.name]: Button,
  },
};
`;


it('自动导入', () => {
  makeStore.set('path/to/HelloWord.vue', [
    "style",
    "template",
    "div",
    "a-form",
    "a-form-item",
    "a-input",
    "a-select",
    "a-select-option",
    "a-button",
    "el-select",
    "el-option",
    "el-input-number",
    "a-config-provider",
    "a-icon",
    "a-space",
    "a-page-header",
    "a-descriptions",
    "a-descriptions-item",
    "a",
    "br",
    "a-tag",
    "a-input-number",
    "a-row",
    "a-statistic",
    "a-form-model",
    "a-form-model-item",
    "a-date-picker",
    "a-switch",
    "a-checkbox-group",
    "a-checkbox",
    "a-radio-group",
    "a-radio",
    "script",
  ]);

  const { code } = babel.transform(example, {
    filename: 'path/to/HelloWord.vue',
    sourceType: 'module',
    plugins: [[plugin, {
      lib(tag) {
        // 如果某个标签需要自动导入，请返回导入路径, 不需要则返回null
        if (tag.startsWith('el-')) {
          return `element-ui/lib/${tag.replace('el-', '')}`;
        }
        if (tag.startsWith('a-')) {
          return `ant-design-vue/lib/${tag.replace('a-', '')}`;
        }
      },
      style(tag) {
        // 如果某个标签需要自动样式文件，请返回导入路径，无则返回null
        if (tag.startsWith('el-')) {
          const label = tag.replace('el-', '');
          return `element-ui/lib/theme-chalk/${label}.css`;
        }
        if (tag.startsWith('a-')) {
          const tagName = tag.replace('a-', '');
          return `ant-design-vue/lib/${tagName}/style`;
        }
        return null;
      },
    }]],
  });
  const resetCode = code.replace(/".*babel-plugin-vue-import-by-tag/g, '"babel-plugin-vue-import-by-tag');
  expect(resetCode).toMatchSnapshot();
});