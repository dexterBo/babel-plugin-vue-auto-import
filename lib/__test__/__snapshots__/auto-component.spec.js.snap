// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`自动导入 1`] = `
"import _default from \\"babel-plugin-vue-import-by-tag/lib/runtime/register.js\\";
import { Button } from 'element-ui';
const _component = {
  name: 'About',
  components: {
    [Button.name]: Button
  }
};
const components = _component.components || {};

_default([,,,,,,,,,,,,,,,,,,,,,,,,,,], components);

_component.components = components;
export default _component;"
`;
